sdv/constraints/base.py

```diff
@@ -220,6 +225,31 @@ def _sample_constraint_columns(self, table_data):
         sampled_data = pd.concat(all_sampled_rows, ignore_index=True)
         return sampled_data
 
+    def _validate_data_on_constraint(self, table_data):
```

Comment:
We also need to check this during `fit`

Reply:
This is a little complicated because (1) `fit` is only really used when `fit_column_model == True`, otherwise all `fit` does is call `_fit`, (2) `_fit` needs to run before we can call `is_valid`, since `_fit` creates some of the attributes used in `is_valid` for some constraints, (3) `_validate_data_on_constraint` runs `is_valid`.

I could (1) move `_validate_data_on_constraint` to run in `fit` right after the `_fit` runs, (2) remove it from the `transform` part, otherwise we would just be running the validation on the same data twice and (3) add it back to the `Table.transform` method.

The reason I didn't do that is because we discussed removing the `fit_columns_model` logic, in which case the logic of both cases is the same, but cleaner the way I did.

Reply:
ok, this is fine for now. The goal eventually is to separate out all of the transformations into their own component. If this happens, it may be possible for users to fit on one dataset and end up using a different one to be transformed, in which case we would have to validate both times. For now though it's fine to just do it in one place